cohttp-eio/src/cohttp_eio.mli

```diff
@@ -56,6 +77,19 @@ module Server : sig
       @raise Invalid_argument
         if [Transfer-Encoding] header in [headers] is not specified as "chunked" *)
 
+  val reader : t -> Eio.Buf_read.t
+  (** [reader t] provides direct access to the request flow.
+
+      This may be useful when switching to another protocol. *)
+
+end
+
+(** [Server] is a HTTP 1.1 server. *)
+module Server : sig
+  type middleware = handler -> handler
+  and handler = Request.t -> response
+  and response = Http.Response.t * Body.t
```

Comment:
why did you make these types mutually recursive?

Reply:
The `middleware` alias isn't used anywhere. Why do we need it?

Reply:
They were already mutually recursive, and middleware was already there. However, I've updated this PR to fix those problems too.